src/NzbDrone.Api/Authentication/EnableAuthInNancy.cs

```diff
@@ -31,5 +53,40 @@ private Response RequiresAuthentication(NancyContext context)
 
             return response;
         }
+
+        private void RegisterFormsAuth(IPipelines pipelines)
+        {
+            if (_configFileProvider.AuthenticationMethod != AuthenticationType.Forms)
+            {
+                return;
+            }
+
+            var salt = new byte[] {1, 2, 3, 4, 5, 6, 7, 8};
+
+            var cryptographyConfiguration = new CryptographyConfiguration(
+                    new RijndaelEncryptionProvider(new PassphraseKeyGenerator(_configService.RijndaelPassphrase, salt)),
+                    new DefaultHmacProvider(new PassphraseKeyGenerator(_configService.HmacPassphrase, salt))
+                );
+
+            FormsAuthentication.Enable(pipelines, new FormsAuthenticationConfiguration
+            {
+                RedirectUrl = "~/login",
+                UserMapper = _authenticationService,
+                CryptographyConfiguration = cryptographyConfiguration
+            });
+        }
+
+        private void RemoveLoginHooksForApiCalls(NancyContext context)
+        {
+            if (context.Request.IsApiRequest())
+            {
+                if ((context.Response.StatusCode == HttpStatusCode.SeeOther && context.Response.ContentType == "text/html") ||
+                    context.Response.StatusCode == HttpStatusCode.Unauthorized)
+                {
+                    context.Response = new { Error = "Unauthorized" }.AsResponse(HttpStatusCode.Unauthorized);
+                    context.Response.Headers.Remove("WWW-Authenticate");
```

Comment:
is this needed? you are replacing the whole response object on the previous line.